refactor ClientQuerySegmentWalker

jisookim0513 opened the pull request:
I have been working on using Spark for long-running Druid queries. I was able to run a long-running druid query without timeout by having intermediate-level brokers that process the query for a specific chunk of segments and letting the final broker merge the query results from intermediate brokers. While doing so, I had to write my own query runners where one does stuff from `RetryQueryRunner` to `toolChest.mergeResults` and the other does stuff from `toolChest.mergeResults` to `FinalizeResultsQueryRunner`. @xvrl suggested refactoring `ClientQuerySegmentWalker` this way so I can use the methods from a new class instead of reusing the code from `ClientQuerySegmentWalker`.

navis commented on server/src/main/java/io/druid/server/FluentQueryRunnerBuilder.java:
```diff
@@ -0,0 +1,123 @@
+package io.druid.server;
```
missing header

CHOIJAEHONG1 commented:
This looks great.

xvrl commented:
I'm of course 👍 on this one, but since I was involved I'd prefer if we get two other +1 on this one

drcrallen commented:
:+1:  needs 1 more

b-slim commented:
LGTM, It will be nice if we add some unit tests to cover the new class.
BTW IT is not a blocking issue.

nishantmonu51 commented:
👍 , LGTM, @jisookim0513: minor nit, could you also modify QueryRunnerTestHelper.makeXXXQueryRunner to use the new api  ?

jisookim0513 commented:
@nishantmonu51 sure, I will modify QueryRunnerTestHelper to use the new API.

gianm commented:
👍

jisookim0513 commented:
@nishantmonu51: modified QueryRunnerTestHelper and refactored FluentQueryRunnerBuiler not to take query as a constructor argument (also moved it to druid.io.processing from druid.io.server)

xvrl commented:
👍 on the latest changes.
